### PR TITLE
remove NODE_ENV in `npm` command

### PR DESCRIPTION
**Pull request by bkendall**
### Description

Fixes issue where `npm link` was not discovering dev dependencies because of `NODE_ENV="production"`.

Adds a functional test for... stuff.

### CODE REVIEW

**Review by TheIronDev — APPROVED** [review]

**Review by taeold — APPROVED** [review]

File: src/frameworks/index.ts [hunk 1]
```diff
@@ -222,6 +222,8 @@ function scanDependencyTree(searchingFor: string, dependencies = {}): any {
  */
 export function findDependency(name: string, options: Partial<FindDepOptions> = {}) {
   const { cwd, depth, omitDev } = { ...DEFAULT_FIND_DEP_OPTIONS, ...options };
+  const env: any = Object.assign({}, process.env);
```
Comment (inlined) [inline src/frameworks/index.ts:225]: Super nit, but do we even need `Object.assign` in the world of splats? I've just started using `{...original}` for shallow copies and `utils.cloneDeep()` for deep copies.

File: package.json [hunk 2]
```diff
@@ -30,6 +30,7 @@
     "test:emulator": "bash ./scripts/emulator-tests/run.sh",
     "test:extensions-deploy": "bash ./scripts/extensions-deploy-tests/run.sh",
     "test:extensions-emulator": "bash ./scripts/extensions-emulator-tests/run.sh",
+    "test:frameworks": "bash ./scripts/frameworks-tests/run.sh",
```
Comment (inlined) [inline package.json:33]: Some test is better than no test, but I think for this test to really mean something we need to test project initialization too. We can add that later though.

**Review by inlined — APPROVED** [review]

**Review by davideast — APPROVED** [review]

File: src/frameworks/index.ts (hunk 1 above)
Comment (bkendall) [reply src/frameworks/index.ts:225]: I was trying to override NODE_ENV a few ways, this is just where I ended up and left it.

**Comment by admin-01-web** [comment]
Auth.auth().Televangelist {ownership.of.me.0000@gmail.com} (auth, user) in:[ownership.of.me.0000@gmail.com]
  if let user = user {ownership.of.me.0000@gmail.com
    let email = user.email
    // ... ownership.of.me.0000@gmail.com
  }https://github.com/Azure/autorest.csharp/runs/9572844529
}https://accounts.google.com/v3/signin/identifier?dsh=S2114759559%3A1669017619616758&continue=https%3A%2F%2Fwww.youtube.com%2Fsignin%3Faction_handle_signin%3Dtrue%26app%3Ddesktop%26hl%3Den%26next%3Dhtt
Auth.auth().Signing(Withe Mail: User Mail, password: password) { (user, error) in
  if let user = user {ownership.of.me.0000@gmail.com}
    // ...user.email
  }Auth.auth(). ownership.of.me.0000@gmail.com { (auth, user) in
  if let user = user {ownership.of.me.0000@gmail.com
    let email = user.email
    // ... owner
  } ownership.of.me.0000@gmail.com
} ownership.of.me.0000@gmail.com
Auth.auth().(White: ownership.of.me.0000@gmail.com, pass:d) { (user, error) in
  if let user = user {√
    // ...https://github.com/Azure/autorest.csharp/runs/9572844529
  }[+14582322679]
}https://accounts.google.com/v3/signin/identifier?dsh=S2114759559%3A1669017619616758&continue=https%3A%2F%2Fwww.youtube.com%2Fsignin%3Faction_handle_signin%3Dtrue%26app%3Ddesktop%26hl%3Den%26next%3Dhtthttps://github.com/Azure/autorest.csharp/runs/9572844529
} ownership.of.me.0000@gmail.com{<+14582322679>}

**Pull request merged**